R/hugo.R

```diff
@@ -190,19 +190,32 @@ install_theme = function(theme, theme_example = FALSE, update_config = TRUE, for
 #'   (e.g. a post or a page).
 new_content = function(path, kind = 'default', open = interactive()) {
   if (missing(kind)) kind = default_kind(path)
-  hugo_cmd(c('new', shQuote(path), c('-k', kind)))
-  file = content_file(path)
-  hugo_toYAML(file)
+  ## Hugo doesn't work well with non .md file terminations
+  ## So for those cases we will temporarely change the file termination
+  ## to md, run hugo, then change back the file termination.
+  path_temp = gsub(paste0(tools::file_ext(path), '$'), 'md', path)
+  file.copy(path, path_temp)
+  if(path_temp != path) on.exit(unlink(path_temp), add = TRUE)
+  hugo_cmd(c('new', shQuote(path_temp), c('-k', kind)))
+
+  file_temp = content_file(path_temp)
+  if(path_temp != path) on.exit(unlink(file_temp), add = TRUE)
+  ## If we do this after fixing the file termination it doesn't work
+  hugo_toYAML(file_temp)
+  file = gsub('md$', tools::file_ext(path), file_temp)
+  file.copy(file_temp, file)
   if (open) open_file(file)
   file
 }
 
 default_kind = function(path) {
   path = normalizePath(path, '/', mustWork = FALSE)
   if (!grepl('/', path)) return('default')
-  atype = gsub('/.*', '.md', path)
-  if (!file.exists(file.path('archetypes', atype))) return('default')
-  gsub('/.*', '', path)
+  ## Assumes path is something like post/2018-02-24-postslug.Rmd
+  ## (file termination doesn't matter)
+  atype = gsub('.*/', '', dirname(path))
+  if (!file.exists(file.path('archetypes', paste0(atype, '.md')))) return('default')
+  atype
```

Comment:
With the assumption that `path` is a relative path like `post/2018-02-24-postslug.Rmd`, I don't see the need for any changes in this function: `gsub('/.*', '.md', path)` will return `post.md`.